### PR TITLE
add 闭环

### PR DESCRIPTION
**Pull request by flaneur2020**
- [x] add token about the 闭环 and lexer
- [x] add parser about the 闭环 statement 
- [x] refactor: take reference instead of move on iterating the ast nodes
- [x] add the interpreter logic about the 闭环 statement
- [ ] support 破圈(break) statement (maybe we could implement it in another PR)

### CODE REVIEW

File: src/evaluator/mod.rs [hunk 1]
```diff
@@ -321,13 +325,30 @@ impl Evaluator {
...
+        &mut self,
+        cond: &Expr,
+        consequence: &BlockStmt,
+    ) -> Option<Object> {
+        let cond = match self.eval_expr(cond) {
+            Some(cond) => cond,
+            None => return None,
+        };
+
+        while Self::is_truthy(cond.clone()) {
```
Comment (chenhengqi) [inline src/evaluator/mod.rs:338]: I am confused about this logic, `cond` is immutable, so we have an infinite loop when it evaluates to `true`, right ?

**Comment by chenhengqi** [comment]
BTW, please add some unit tests for this feature.

File: src/evaluator/mod.rs (hunk 1 above)
Comment (flaneur2020) [reply src/evaluator/mod.rs]: yeah it's a bug, maybe it'd been fixed on my latest commit

**Comment by flaneur2020** [comment]
> BTW, please add some unit tests for this feature.

thank your for remind, i've add some test cases about this, please check again 🤝

**Comment by chenhengqi** [comment]
LGTM

File: src/evaluator/mod.rs [hunk 2]
```diff
@@ -321,13 +325,35 @@ impl Evaluator {
         }
     }
 
-    fn eval_call_expr(&mut self, func: Box<Expr>, args: Vec<Expr>) -> Object {
+    fn eval_while_expr(
```
Comment (Artoria2e5) [inline src/evaluator/mod.rs:328]: 解 Box 好啊！clippy 总跟我叨叨。

**Review by Artoria2e5 — APPROVED** [review]

File: src/evaluator/mod.rs (hunk 2 above)
Comment (flaneur2020) [reply src/evaluator/mod.rs:328]: qwq！

**Review by chenhengqi — APPROVED** [review]

**Pull request merged**